lesson4/argirovatina_hometask.rb

```diff
@@ -0,0 +1,45 @@
+class LazyHash < Hash
+
+  def method_missing(method_name)
+    method_name = get_correct_key(method_name)
+    if key_is_duplicated?(method_name)
+      raise "Keys #{method_name} as string and as symbol are present"
+    elsif self.keys.include? method_name
```

Comment:
You don't need to check it again.
You already get a proper key by `#get_correct_key` if it present. If it isn't, `#get_correct_key` should return `nil`. Check this result.